VTUL/dlp-access
# Added sponsors configurations

**JIRA Ticket**: (link) (:star:)
https://webapps.es.vt.edu/jira/browse/LIBTD-2342

# What does this Pull Request do? (:star:)
Adds fields to configure sponsors data.

# What's the changes? (:star:)
I changed the HomepageTop component to Homepage and added the form fields and viewing fields for the sponsors to this component. Users can edit the alt text, and link for each sponsor, upload a new image for an existing sponsor (note: file name must match exactly what is shown in "source" - including the file extension). Users can add a new sponsor with image, or delete a sponsor. Also edited the SiteSponsors component to not display if array is present but empty. A cypress test is included.

# How should this be tested?
Log in to the siteAdmin page. Click on "Hompage config" and then choose edit. Edit existing sponsors data, upload a new image for a sponsor, or add a new sponsor.  Click update config. The new sponsor information should appear on the homepage.

# Additional Notes:
Branch is LIBTD-2343

# Interested parties
@yinlinchen 

(:star:) Required fields
